### PR TITLE
Audio: up_down_mixer: move up_down_mixer header files to module folder

### PR DESCRIPTION
**Pull request by btian1**
This is a clean up, purpose is de-cluster headers, toml files,
Readme.md etc per module basis, since today everything is scattered
in current code base.

### CODE REVIEW

**Review by RanderWang — COMMENTED** [review]
Can you explain why we need to do it ? Thanks!

File: src/audio/up_down_mixer/coef/up_down_mixer.h [hunk 1]
```diff
@@ -5,7 +5,7 @@
 // Author: Bartosz Kokoszko <bartoszx.kokoszko@intel.com>
 // Author: Adrian Bonislawski <adrian.bonislawski@intel.com>
 
-#include <ipc4/up_down_mixer.h>
+#include "../up_down_mixer_ipc4.h"
```
Comment (lyakh) [inline src/audio/up_down_mixer/coef/up_down_mixer.h:8]: can we just name this file "coef.h" and place it at the same level?

Comment (btian1) [reply src/audio/up_down_mixer/coef/up_down_mixer.h:8]: this is not coef header, coef header put it to coef/up_down_mixer.h
this header have definition:
ipc4_up_down_mixer_module_cfg

**Comment by btian1** [comment]
> Can you explain why we need to do it ? Thanks!

purpose is all module content in the same folder: header/source/kconfig/CMakeList, etc, this is part of our Q4 OKR.

File: src/audio/up_down_mixer/coef/up_down_mixer.h (hunk 1 above)
Comment (lyakh) [reply src/audio/up_down_mixer/coef/up_down_mixer.h:8]: yes, I mean the coefficient header, i.e. the file, which is being patched here, not the one, included by this instruction, sorry for the confusion

Comment (btian1) [reply src/audio/up_down_mixer/coef/up_down_mixer.h:8]: Do you mean change coef/up_down_mixer.h --> up_down_mixer_coef.h ?

Comment (btian1) [reply src/audio/up_down_mixer/coef/up_down_mixer.h:8]: it is not coef.h, it is in coef folder with name up_down_mixer.h. 
downmix only have one header file, however, for src/asrc, we have hundred of header files, change file name is huge cost.
how about keep it with coef/up_down_mixer.h?

Comment (lyakh) [reply src/audio/up_down_mixer/coef/up_down_mixer.h:8]: > it is not coef.h, it is in coef folder with name up_down_mixer.h. downmix only have one header file, however, for src/asrc, we have hundred of header files, change file name is huge cost. how about keep it with coef/up_down_mixer.h?

@btian1 I know that your current version calls it "coef/up_down_mixer.h" and that's what I'm proposing to change for two reasons: (1) two files with the same name in adjacent directories look confusing, (2) no need to create an additional directory level just for one file. So, originally I proposed to just do
```
git mv coef/up_down_mixer.h coef.h
```
to which you proposed to call it up_down_mixer_coef.h instead which is also fine with me, but not using the same name as in your current version.

Comment (btian1) [reply src/audio/up_down_mixer/coef/up_down_mixer.h:8]: ok, use up_down_mixer_coef.h is better since coef.h is too common for file name.

**Review by lgirdwood — APPROVED** [review]

**Comment by kv2019i** [comment]
@btian1 Build errors in CI, please check, e.g https://github.com/thesofproject/sof/actions/runs/7086932318/job/19286139196?pr=8552

**Review by lyakh — APPROVED** [review]

**Pull request merged**